src/utilities/effects.ts

```diff
@@ -1069,64 +1069,144 @@ const effects = {
         const subtreeDeletions = sortedActions[ActivityDeletionAction.DELETE_CHAIN] ?? [];
         const normalDeletions = sortedActions[ActivityDeletionAction.NORMAL] ?? [];
 
+        // The following deletion queries must occur in a specific order to avoid errors from deleting
+        // directives that still have other activities dependent on them
         if (reanchorRootDeletions.length) {
-          await reqHasura(
+          const response = await reqHasura<
+            {
+              affected_row: ActivityDirective;
+              change_type: string;
+            }[]
+          >(
             gql.DELETE_ACTIVITY_DIRECTIVES_REANCHOR_TO_ANCHOR,
             {
               activity_ids: convertToGQLArray(reanchorRootDeletions),
               plan_id,
             },
             user,
           );
-          activityDirectivesMap.update((currentActivityDirectivesMap: ActivityDirectivesMap) => {
-            reanchorRootDeletions.forEach(id => delete currentActivityDirectivesMap[id]);
-            return { ...currentActivityDirectivesMap };
-          });
+
+          if (response.delete_activity_by_pk_reanchor_to_anchor_bulk != null) {
+            const deletedActivityIds = response.delete_activity_by_pk_reanchor_to_anchor_bulk
+              .filter(({ change_type }) => {
+                return change_type === 'deleted';
+              })
+              .map(({ affected_row: { id } }) => id);
+
+            activityDirectivesMap.update((currentActivityDirectivesMap: ActivityDirectivesMap) => {
+              deletedActivityIds.forEach(id => delete currentActivityDirectivesMap[id]);
+              return { ...currentActivityDirectivesMap };
+            });
+
+            // If there are activities that did not get deleted
```

Comment:
Any easy way to test this?

Reply:
> Any easy way to test this?

I don't know of an easy way unfortunately, but in a followup PR, I intend on at least adding unit tests for this.